src/Uart/Uart.c

```diff
@@ -314,24 +319,40 @@ handleTxInterrupt(Uart *const uart)
 	}
 }
 
-void
-Uart_handleInterrupt(Uart *const uart)
+static inline bool
+Uart_hasAnyErrorOccured(Uart_ErrorFlags* const errFlags)
 {
-	uint32_t status = uart->reg->sr;
+	return (errFlags->hasFramingErrorOccurred || errFlags->hasOverrunOccurred || errFlags->hasParityErrorOccurred
+            || errFlags->hasRxFifoFullErrorOccurred);
+}
+
+bool
+Uart_handleInterrupt(Uart *const uart, int * errCode)
+{
+	bool retValue = true;
+	Uart_ErrorFlags errorFlags = { false, false, false, false };
+
+	uint32_t status = uart->reg->sr & uart->reg->imr;
 	uart->reg->cr = UART_CR_RSTSTA_MASK;
 	if ((status & UART_SR_RXRDY_MASK) != 0u)
-		handleRxInterrupt(uart);
+	{
+		retValue &= handleRxInterrupt(uart, errCode);
```

Comment:
Now we have a problem. If the parent passes NULL (and it does in the runtime....), then this error is not reported in the error callback, right?

Reply:
I believe that we could have a local errCode, from which the error could be propagated to the parameter, if not Null

Reply:
We could also resign from passing a reference to error code at all. Why not handle all of the errors just in the error handler instead of passing them outside.